datafusion/expr/src/udaf.rs

```diff
@@ -360,6 +391,27 @@ pub trait AggregateUDFImpl: Debug + Send + Sync {
         &[]
     }
 
+    /// Sets the flag specifying whether the requirement of the UDF is satisfied.
```

Comment:
I found this name / documenation to be a little confusing - specifically it isn't clear from just this documentation (the public API of `AggregageUDFImpl` *what* type of requirement is being satisfied. 

The documentation on AggregateExpr makes it clearer -- maybe we could find a way to bring some of that description to this location, or link to there for more details

Also I think we could improve the name of this function

1.  `with_...` is often used for builder style APIs in rust, which this is not
2. It uses "requirement" which is general 


Maybe it would be clearer it we called it `has_beneficial_ordering` or `input_ordered_correctly` 🤔